docs/source/configuration/config_loader_migration.md

```diff
@@ -0,0 +1,65 @@
+This migration guide outlines the primary distinctions between the old loaders and the `OmegaConfigLoader`, providing step-by-step instructions on updating your code base to utilise the new class effectively.
+
+## [`ConfigLoader`](/kedro.config.ConfigLoader) to [`OmegaConfigLoader`](/kedro.config.OmegaConfigLoader)
+
+### 1. Install the Required Library
+The [`OmegaConfigLoader`](advanced_configuration.md#omegaconfigloader) was introduced in Kedro `0.18.5` and is based on [OmegaConf](https://omegaconf.readthedocs.io/). In order to use it you need to ensure you have both a version of Kedro of `0.18.5` or above and `omegaconf` installed.
+You can install both using `pip`:
+
+```bash
+pip install kedro==0.18.5
+```
+This would be the minimum required Kedro version which includes `omegaconf` as dependency.
+Or you can run:
+```bash
+pip install -U kedro
+```
+
+This command installs the most recent version of Kedro which also includes `omegaconf` as dependency.
+
+### 2. Use the `OmegaConfigLoader`
+To use `OmegaConfigLoader` in your project, set the `CONFIG_LOADER_CLASS` constant in your [`src/<package_name>/settings.py`](../kedro_project_setup/settings.md):
+
+```python
+from kedro.config import OmegaConfigLoader  # new import
+
+CONFIG_LOADER_CLASS = OmegaConfigLoader
+```
+
+### 3. Import Statements
+Replace the import statement for `ConfigLoader` with the one for `OmegaConfigLoader`:
+
+```python
+# Before:
+from kedro.config import ConfigLoader
+
+# After:
+from kedro.config import OmegaConfigLoader
+```
+
+### 4. File Format Support
+`OmegaConfigLoader` supports only `yaml` and `json` file formats. Make sure that all your configuration files are in one of these formats. If you were using other formats with `ConfigLoader`, convert them to `yaml` or `json`.
+
+### 5. Load Configuration
+The method to load the configuration using `OmegaConfigLoader` differs slightly from that used by `ConfigLoader`, which allowed users to access configuration through the `.get()` method and required patterns as argument.
+When you migrate to use `OmegaConfigLoader`it  requires you to fetch configuration through a configuration key that points to [configuration patterns specified in the loader class](configuration_basics.md#configuration-patterns) or [provided in the `CONFIG_LOADER_ARGS`](advanced_configuration.md#how-to-change-which-configuration-files-are-loaded) in `settings.py`.
+
+```python
+# Before:
+conf_path = str(project_path / settings.CONF_SOURCE)
+conf_loader = ConfigLoader(conf_source=conf_path, env="local")
+catalog = conf_loader.get("catalog*")
+
+# After:
+conf_path = str(project_path / settings.CONF_SOURCE)
+config_loader = OmegaConfigLoader(conf_source=conf_path, env="local")
+catalog = config_loader["catalog"]
```

Comment:
Isn't the pattern and dictionary interface available for all config loader? Maybe I remember the wrong thing.

Reply:
Yes it is. I put this in because currently both `.get("catalog*")` and `[catalog]` works for `ConfigLoader` but for `OmegaConfigLoader` only `[catalog]` works. I don't know if users have already switched to the newer dictionary interface.